pandas/core/arrays/datetimes.py

```diff
@@ -152,6 +162,10 @@ class DatetimeArrayMixin(dtl.DatetimeLikeArrayMixin):
                  'is_year_end', 'is_leap_year']
     _object_ops = ['weekday_name', 'freq', 'tz']
 
+    # dummy attribute so that datetime.__eq__(DatetimeArray) defers
```

Comment:
huh? why don't you just implement and raise NotIMplemented?

Reply:
This is is a quirk of the stdlib `datetime` implementation.  `datetime == other` will return `NotImplemented` instead of raising `TypeError` iff `other` has a `timetuple` attribute.